.github/workflows/master.yml

```diff
@@ -0,0 +1,83 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 Canonical Ltd.
+
+name: Master workflow
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Build
+        run: go build
+
+  docker-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Build and push Docker image
+        run: make docker-build
+
```

Comment:
```suggestion
```

This project does not create a docker image.

Reply:
It does create a Docker image but not for it root directory. It is inside `mongoapi/dbtestapp/`

Reply:
dbtestapp directory is indicated.

Reply:
This is a way how I made it work (add `go.work`). However, I would prefer to avoid this "solution" because we are not using `go.work` in any of our repos

```diff
diff --git a/go.work b/go.work
new file mode 100644
index 0000000..38b810b
--- /dev/null
+++ b/go.work
@@ -0,0 +1,6 @@
+go 1.21
+
+use (
+  ./
+  ./mongoapi/dbtestapp
+)
diff --git a/go.work.sum b/go.work.sum
new file mode 100644
index 0000000..6060b61
--- /dev/null
+++ b/go.work.sum
@@ -0,0 +1,11 @@
+github.com/antonfisher/nested-logrus-formatter v1.3.1 h1:NFJIr+pzwv5QLHTPyKz9UMEoHck02Q9L0FP13b/xSbQ=
+github.com/cespare/xxhash v1.1.0 h1:a6HrQnmkObjyL+Gs60czilIUGqrzKutQD6XZog3p+ko=
+github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e h1:Wf6HqHfScWJN9/ZjdUKyjop4mf3Qdd+1TvvltAvM3m8=
+github.com/go-redis/redis v6.15.9+incompatible h1:K0pv1D7EQUjfyoMql+r/jZqCLizCGKFlFgcHWWmHQjg=
+github.com/jtolds/gls v4.20.0+incompatible h1:xdiiI2gbIgH/gLH7ADydsJ1uDOEzR8yvV7C0MuV77Wo=
+github.com/mitchellh/mapstructure v1.4.1 h1:CpVNEelQCZBooIPDn+AR3NpivK/TIKU8bDxdASFVQag=
+github.com/pelletier/go-toml v1.9.3 h1:zeC5b1GviRUyKYd6OJPvBU/mcVDVoL1OhT17FCt5dSQ=
+github.com/smartystreets/assertions v0.0.0-20180927180507-b2de0cb4f26d h1:zE9ykElWQ6/NYmHa3jpm/yHnI4xSofP+UP6SpjHcSeM=
+github.com/smartystreets/goconvey v1.6.4 h1:fv0U8FUIMPNf1L9lnHLvLhgicrIVChEkdzIKYqbNC9s=
+github.com/ugorji/go v1.2.7 h1:qYhyWUUd6WbiM+C6JZAUkIJt/1WrjzNHY9+KCIjVqTo=
+go.etcd.io/etcd v0.5.0-alpha.5.0.20200910180754-dd1b699fc489 h1:1JFLBqwIgdyHN1ZtgjTBwO+blA6gVOmZurpiMEsETKo=
```

So, from the above, it looks tom me like something, such as `mongoapi`, is not properly imported.

Using the `go.work` file, this is what I get when trying to build the code:
```bash
:~/.../mongoapi/dbtestapp$ go build ./...
# testapp
./drsmapp.go:55:18: cannot use drsm.InitDRSM(resName, podId, db, opt) (value of type drsm.DrsmInterface) as *drsm.Drsm value in assignment: need type assertion
```

Reply:
I think, the problem occurs because of this content https://github.com/omec-project/util/blob/86f382d925908271b110c580dc1c73ee386543e2/mongoapi/dbtestapp/go.mod#L10C1-L10C45. We can use the latest tag `github.com/omec-project/util v1.0.11`, but that does not solve the issue as it does not include the API changes. We need to create another tag which includes the final changes.

Reply:
Could you please help to create a new tag ?

Reply:
`./drsmapp.go:55:18: cannot use drsm.InitDRSM(resName, podId, db, opt) (value of type drsm.DrsmInterface) as *drsm.Drsm value in assignment: need type assertion` is fixed in the latest update.

Reply:
> Could you please help to create a new tag ?

Done!

Reply:
Is this comment/conversation resolved? If so, please mark it as resolved